### ISSUE TITLE
MPA Stability Fee

### ISSUE THREAD

**Issue by abitmore** [bitshares/bitshares-core#2361]
**User Story**

As a MPA owner I'd like to be able to set a stability fee on my MPA so that the business is more sustainable. The fee works like an interest for borrowing the debt asset.

Key features:
* debt position holders pay a percentage fee periodically, automatically
* if a debt position's collateral ratio (CR) is too low after paying the fee, it will get margin called
  *  in extreme circumstances it may cause a black swan event
* fee rate is adjustable by asset owners if certain issuer permission / flag is set

Optional features (perhaps add later):
* distribute a percentage of the stability fee to debt asset holders

**Impacts**
Describe which portion(s) of BitShares Core may be impacted by your request. Please tick at least one box.
- [ ] API (the application programming interface)
- [ ] Build (the build process or something prior to compiled code)
- [ ] CLI (the command line wallet)
- [ ] Deployment (the deployment process after building such as Docker, Travis, etc.)
- [ ] DEX (the Decentralized EXchange, market engine, etc.)
- [ ] P2P (the peer-to-peer network for transaction/block propagation)
- [ ] Performance (system or user efficiency, etc.)
- [x] Protocol (the blockchain logic, consensus, validation, etc.)
- [ ] Security (the security of system or user data, etc.)
- [ ] UX (the User Experience)
- [ ] Other (please add below)

## CORE TEAM TASK LIST
- [ ] Evaluate / Prioritize Feature Request
- [ ] Refine User Stories / Requirements
- [ ] Define Test Cases
- [ ] Design / Develop Solution
- [ ] Perform QA/Testing
- [ ] Update Documentation

**Comment by shulthz** [comment]
I don't think this "debt position holders pay a percentage fee **periodically**" is suitable, 

but it can be designed to charge a percentage fee when the debt position holders close his position.
